### ISSUE TITLE
Sometimes harbour-callrecorderd starts to constantly use 100% of CPU

### ISSUE THREAD

**Issue by Self-Perfection** [dpurgin/harbour-callrecorder#41]
It seems that it somehow related to phone call events. Dunno whether it is caused by answering call, stopping call or maybe during call. A bit of diagnostis follows.

Hangs just one thread (see light weight process (thread) ID 4143):

```
$ ps -FLC harbour-callrecorderd
UID        PID  PPID   LWP  C NLWP    SZ   RSS PSR STIME TTY          TIME CMD
nemo      3989   863  3989  0    9 91611 12632   0 Nov02 ?        00:01:20 /usr/bin/harbour-callrecorderd
nemo      3989   863  4010  0    9 91611 12632   1 Nov02 ?        00:00:00 /usr/bin/harbour-callrecorderd
nemo      3989   863  4141  0    9 91611 12632   1 Nov02 ?        00:00:16 /usr/bin/harbour-callrecorderd
nemo      3989   863  4143  3    9 91611 12632   0 Nov02 ?        01:56:34 /usr/bin/harbour-callrecorderd
nemo      3989   863  4144  0    9 91611 12632   0 Nov02 ?        00:00:01 /usr/bin/harbour-callrecorderd
nemo      3989   863  4145  0    9 91611 12632   0 Nov02 ?        00:00:00 /usr/bin/harbour-callrecorderd
nemo      3989   863  4146  0    9 91611 12632   1 Nov02 ?        00:00:00 /usr/bin/harbour-callrecorderd
nemo      3989   863  4189  0    9 91611 12632   0 Nov02 ?        00:00:04 /usr/bin/harbour-callrecorderd
nemo      3989   863  4259  0    9 91611 12632   0 Nov02 ?        00:00:00 /usr/bin/harbour-callrecorderd
```

It seems hanged thread constantly tries to read something:

```
$ strace -tt -p 4143 |& head -10
Process 4143 attached
23:26:04.652947 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.654900 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.655052 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.655236 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.655419 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.656334 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.656517 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.656670 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
23:26:04.656853 read(18, 0x4e4ffc58, 10) = -1 EAGAIN (Resource temporarily unavailable)
```

But I don't have any idea what it tries to read:

```
$ ls -l /proc/4143/fd/18
lr-x------ 1 nemo nemo 64 2015-11-04 23:00 /proc/4143/fd/18 -> pipe:[1770970]
```

Such situation happened to me 2 or 3 times. Really annoying to find 2 hours after a call that your device is suddenly almost out of charge.
P.S.: I'm still using 0.6.0 version, just learned there is newer release. But it does not look like 0.6.1 have related fixes.

**Comment by Self-Perfection** [comment]
I have not noticed this issue since I had reported it. I suppose it was fixed by callrecorder or OS update. Let's consider it as resolved.

**Comment by Self-Perfection** [comment]
I observe exactly the same hang right now
```
$ rpm -q harbour-callrecorder
harbour-callrecorder-0.7.0-1.armv7hl
$ ps -FLC harbour-callrecorder
UID        PID  PPID   LWP  C NLWP    SZ   RSS PSR STIME TTY          TIME CMD
nemo     21514   861 21514  0    3 82032  7232   0 01:42 ?        00:00:29 /usr/bin/harbour-callrecorderd
nemo     21514   861 21527  0    3 82032  7232   0 01:42 ?        00:00:04 /usr/bin/harbour-callrecorderd
nemo     21514   861 21531 17    3 82032  7232   1 01:42 ?        02:59:45 /usr/bin/harbour-callrecorderd
```

**Comment by dpurgin** [comment]
That's odd. Are there any visible symptoms? Can you please upload the log from `/home/nemo/.local/share/kz.dpurgin/harbour-callrecorder/log.txt` or send it to me per email? It contains non-masked phone numbers though, so it might be privacy-sensitive.

**Comment by Self-Perfection** [comment]
I've sent you an e-mail with log fragment with events around the time of hanging. At 18:38 I got incoming call. When I reached my phone I noticed that it was hot. Luckily it was attached to PC for hours thus it was not discharged.

**The call at 18:48 was recorded without sound of remote side** (I hear only my voice and surrounding noise in the recording).

Consumed CPU time by daemon process suggests, that something went wrong at the time of previous call and `harbour-callrecorderd` was "hanging" since that moment.

-

Offtopic: while I was composing this reply I received my brand new Jolla C. Happy me. Probably OS call recording function would be enough for me and I won't use `harbour-callrecorder` on new device.
